core/src/main/java/org/elasticsearch/action/admin/indices/rollover/TransportRolloverAction.java

```diff
@@ -111,42 +110,45 @@ public void onResponse(IndicesStatsResponse statsResponse) {
-                        createIndexService.createIndex(prepareCreateIndexRequest(rolloverIndexName, rolloverRequest),
-                            new ActionListener<ClusterStateUpdateResponse>() {
-                                @Override
-                                public void onResponse(ClusterStateUpdateResponse response) {
-                                    // switch the alias to point to the newly created index
-                                    indexAliasesService.indicesAliases(
-                                        prepareRolloverAliasesUpdateRequest(sourceIndexName, rolloverIndexName,
-                                            rolloverRequest),
-                                        new ActionListener<ClusterStateUpdateResponse>() {
-                                            @Override
-                                            public void onResponse(ClusterStateUpdateResponse clusterStateUpdateResponse) {
-                                                listener.onResponse(
-                                                    new RolloverResponse(sourceIndexName, rolloverIndexName,
-                                                        conditionResults, false, true));
-                                            }
-
-                                            @Override
-                                            public void onFailure(Throwable e) {
-                                                listener.onFailure(e);
-                                            }
-                                        });
-                                }
-
-                                @Override
-                                public void onFailure(Throwable t) {
-                                    listener.onFailure(t);
-                                }
-                            });
+                        CreateIndexClusterStateUpdateRequest updateRequest = prepareCreateIndexRequest(rolloverIndexName, rolloverRequest);
+                        ActionListener<RolloverResponse> wrappedListener = new ActionListener<RolloverResponse>() {
+
+                            @Override
+                            public void onResponse(RolloverResponse rolloverResponse) {
+                                // switch the alias to point to the newly created index
+                                indexAliasesService.indicesAliases(
+                                    prepareRolloverAliasesUpdateRequest(sourceIndexName, rolloverIndexName,
+                                        rolloverRequest),
+                                    new ActionListener<ClusterStateUpdateResponse>() {
+                                        @Override
+                                        public void onResponse(ClusterStateUpdateResponse clusterStateUpdateResponse) {
+                                            // TODO: what happens if this ClusterStateUpdateResponse is not acknowledged?
```

Comment:
I also am in favor of the former, I didn't change anything here originally because that is how the rollover action worked.  If it should be changed to do something else on `acknowledged = false`, we can make that in a separate PR?